### PR TITLE
feat(gemini): autogenerated update

### PR DESCRIPTION
**Pull request by polsala**
## Rationale
- Automated proposal from the Gemini generator delivering a fresh community utility.

## Why safe to merge
- Utility is isolated to `utils/<util_name>`.
- README + tests ship together (see folder contents).
- No secrets or credentials touched.

## Test Plan
- Follow the instructions in the generated README

## Links
- Generated docs and examples committed alongside this change.

## Mock Justification
- Not applicable; generator did not introduce new mocks.

### CODE REVIEW

**Comment by polsala** [comment]
✅ What’s solid
*   **Clear Purpose and Scope:** The utility provides a well-defined, self-contained function for offline emoji lookup, which is a useful addition.
*   **No External Dependencies:** The `src/emoji_lookup.py` explicitly states and demonstrates that it has no external dependencies, simplifying deployment and reducing supply chain risk.
*   **Offline Operation:** The use of a curated, in-memory emoji database ensures fast, reliable operation without network calls.
*   **Dual Interface:** The utility offers both a callable `search` function for programmatic use and a command-line interface, providing flexibility for different use cases.
*   **Robust CLI Input Handling:** The `main` function correctly handles keywords provided as arguments or piped via `stdin`, including stripping whitespace and ignoring empty lines.
*   **Type Hinting and Docstrings:** Good use of type hints and clear docstrings enhances code readability and maintainability.

🧪 Tests
*   **Comprehensive Coverage:** The test suite covers the core `search` function's behavior (basic match, substring, case-insensitivity, no match) and the `main` function's CLI interactions (with arguments, reading from stdin, no matches).
*   **Effective CLI Testing:** The use of `pytest` fixtures like `monkeypatch` and `capsys` is appropriate and effective for simulating `sys.argv`, `sys.stdin`, and capturing `sys.stdout`.
*   **Test Setup:** The `sys.path.append` in `test_emoji_lookup.py` correctly ensures the `src` module is discoverable for testing.

**Actionable Feedback:**
*   **Edge Cases for `search` Function:** Add explicit tests for edge cases in the `search` function to confirm expected behavior:
    *   An empty string query: `search("")`
    *   A query consisting only of whitespace: `search("   ")`
    *   A query with leading/trailing whitespace: `search(" smile ")`
    *   A query that is a substring of multiple keywords (e.g., "a" should match "cat", "laugh", "star", "party").
*   **CLI Empty Stdin Test:** While the current implementation correctly handles an empty `stdin` by exiting with code 1, an explicit test case for this scenario would improve coverage.
    ```python
    import io
    import sys
    from src.emoji_lookup import main
    import pytest

    def test_cli_empty_stdin_exits_with_1(monkeypatch):
        monkeypatch.setattr(sys, "stdin", io.StringIO(""))
        monkeypatch.setattr(sys, "argv", ["emoji_lookup"])
        exit_code = main()
        assert exit_code == 1
    ```

🔒 Security
*   **Minimal Attack Surface:** The utility's design with no external dependencies and offline operation significantly reduces potential security vulnerabilities related to supply chain attacks or network-based exploits.
*   **No Credential Handling:** The PR explicitly states no secrets or credentials are touched, which is a strong security posture.
*   **Input Sanitization:** The `keyword.lower().strip()` operation on search queries helps prevent unexpected behavior from malformed input.

**Actionable Feedback:**
*   **Resource Consumption for Large Inputs:** While the current `EMOJI_DB` is small, if it were to grow significantly or if the utility were to process extremely long lines from `stdin`, there could be a theoretical, albeit minor, risk of resource exhaustion (memory/CPU) due to string processing and list extensions. For the current scope, this is not a practical concern, but it's a consideration for future scaling.

🧩 Docs/DX
*   **Comprehensive `README.md`:** The `README.md` is well-structured, providing clear features, installation instructions, and usage examples for both direct arguments and stdin piping.
*   **Developer Experience:** The lack of external dependencies and clear test instructions contribute to a positive developer experience.
*   **CLI Help:** `argparse` automatically provides a helpful `--help` message for the command-line tool.

**Actionable Feedback:**
*   **Nested Directory Structure:** The path `utils/nightly-nightly-emoji-lookup/utils/nightly-emoji-lookup` appears redundant. Consider flattening this to `utils/nightly-emoji-lookup` to simplify navigation and installation instructions.
    *   If the nested structure is intentional (e.g., `nightly-nightly-emoji-lookup` is a generator-specific wrapper), clarify its purpose in the `README` or PR body.
*   **Installation Path in `README`:** The `README.md` currently instructs `cd utils/nightly-emoji-lookup`. This should be updated to reflect the actual nested path or the path after any proposed flattening.
    ```diff
    --- a/utils/nightly-nightly-emoji-lookup/utils/nightly-emoji-lookup/README.md
    +++ b/utils/nightly-nightly-emoji-lookup/utils/nightly-emoji-lookup/README.md
    @@ -14,7 +14,7 @@
     ## Installation
     ```bash
     # From the repository root
    -cd utils/nightly-emoji-lookup
    +cd utils/nightly-nightly-emoji-lookup/utils/nightly-emoji-lookup
     python -m venv .venv
     source .venv/bin/activate
     pip install -e .
    ```
*   **User Installation vs. Dev Install:** The `pip install -e .` command is for "editable" (development) installs. For general users, `pip install .` (or `pip install <path_to_dir>`) is more common for a non-editable installation. Clarify in the `README` if `pip install -e .` is the intended method for all users, or provide an alternative for non-developers.
*   **Standalone CLI Entry Point:** For improved user experience, consider adding a `pyproject.toml` with an `entry_points` section to allow users to run the utility directly as `emoji-lookup smile` after installation, rather than `python -m src.emoji_lookup smile`.
    ```toml
    # In pyproject.toml at the root of the utility
    [project.scripts]
    emoji-lookup = "src.emoji_lookup:main"
    ```
*   **`__future__` Import:** The `from __future__ import annotations` import is not strictly necessary for Python 3.11+ as type hints are non-evaluated by default. It doesn't cause harm but could be removed for conciseness.

🧱 Mocks/Fakes
*   **Accurate Rationale:** The PR body and test file correctly state that no application-level mocks are needed because the utility operates entirely in-memory without external I/O or network calls.
*   **System Interface Mocks:** The tests effectively use `pytest`'s `monkeypatch` and `capsys` fixtures to mock system interfaces (`sys.stdin`, `sys.argv`, `sys.stdout`) for testing the CLI functionality. These are appropriate and well-implemented.

**Actionable Feedback:**
*   **Clarify Mock Justification:** While the current justification is accurate for application logic, explicitly mentioning the use of `monkeypatch` and `capsys` for mocking system interfaces in the PR's "Mock Justification" section would provide a more complete picture of the testing strategy.
    *   **Example for PR Body:**
        ```
        ## Mock Justification
        - Not applicable for application logic; the utility operates entirely in-memory.
        - System interfaces (`sys.stdin`, `sys.argv`, `sys.stdout`) are mocked in tests using `pytest`'s `monkeypatch` and `capsys` fixtures to simulate CLI interactions.
        ```
